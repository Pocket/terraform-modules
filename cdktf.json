{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
