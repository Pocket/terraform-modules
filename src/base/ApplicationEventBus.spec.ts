import { Testing } from 'cdktf';
import {
  ApplicationEventBus,
  ApplicationEventBusProps,
} from './ApplicationEventBus';

describe('ApplicationEventBus', () => {
  it('renders an event bus with name and target', () => {
    const props: ApplicationEventBusProps = {
      name: 'test-event-bus',
      tags: { service: 'test-service' },
    };
    const synthed = Testing.synthScope((stack) => {
      new ApplicationEventBus(stack, 'test-event-bus', props);
    });
    expect(synthed).toMatchSnapshot();
  });
});
