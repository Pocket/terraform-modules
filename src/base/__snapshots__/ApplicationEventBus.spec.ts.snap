// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationEventBus renders an event bus with name and target 1`] = `
"{
  \\"resource\\": {
    \\"aws_cloudwatch_event_bus\\": {
      \\"test-event-bus_event-bus-test-event-bus_3B38546A\\": {
        \\"name\\": \\"test-event-bus\\",
        \\"tags\\": {
          \\"service\\": \\"test-service\\"
        }
      }
    }
  }
}"
`;
